src/components/TileEditor.tsx

```diff
@@ -1,14 +1,78 @@
 import React from 'react';
-import { TileInfo } from '../firebase';
+import { firestore, TileInfo } from '../firebase';
+import { useForm } from "react-hook-form";
 
 export function TileEditor(props: {tileInfo: TileInfo, tileName: string}) {
+  const { register, setValue, handleSubmit, errors } = useForm<TileInfo>();
+  const onSubmit = handleSubmit( async ({ bg_color, bg_image, hover_text, description_text, link }) => {
+    const tile = firestore.collection("tiles").doc(props.tileName);
+    try{const result = await tile.update({ 
+        bg_color: bg_color,
+        bg_image: bg_image,
+        hover_text: hover_text,
+        description_text: description_text,
+        link: link
+      });      
+    } catch {
+      console.log("Update failed")
+    }
+  });
+
   return (
-    <div>
-      <h2>Tile: {props.tileName}</h2>
-      <h3>Hover Text: {props.tileInfo.hover_text}</h3>
-      <h3>Description: {props.tileInfo.description_text}</h3>
-    </div>
+    <form onSubmit={onSubmit}>
+      <h2>{props.tileName}</h2>
+      <label>Color
+        <input 
+          name="bg_color"
+          type="text"
+          key={`bg_color:${props.tileInfo.bg_color}`}
+          defaultValue={props.tileInfo.bg_color}
+          ref={register}
+        />
+      </label>
+      <br />
+      <label>Image
+        <input
+          name="bg_image"
+          type="text"
+          key={`bg_image:${props.tileInfo.bg_image}`}
+          defaultValue={props.tileInfo.bg_image}
+          ref={register}
+        />
+      </label>
+      <br />
+      <label>Hover Text
+        <input
+          name="hover_text"
+          type="text"
+          key={`hover_text:${props.tileInfo.hover_text}`}
+          defaultValue={props.tileInfo.hover_text}
+          ref={register}
+          />
+      </label>
+      <br />
+      <label>Description Text
+        <input
+          name="description_text"
+          type="text"
+          key={`description_text:${props.tileInfo.description_text}`}
+          defaultValue={props.tileInfo.description_text}
+          ref={register}
+        />
+      </label>
+      <br />
+      <label>Link
+       <input
+         name="link"
+         type="text"
+         key={`link:${props.tileInfo.link}`}
+         defaultValue={props.tileInfo.link}
+         ref={register} />
+      </label>
+      <br />
+      <input type="submit" value="Update Tile" />
+    </form>
   );
-}
+};
 
-export default TileEditor;
+export default TileEditor;
```

Comment:
Please run `npm run lint` before opening a PR; this will automatically fix whitespaces and semicolons.

Reply:
Sorry - the convention I'm used to PRs are considered WIP until review is explicitly requested. I'll try to remember to lint stuff before I push it up in the future (and also try to use the draft PRs for WIPs).

Reply:
Ah, that's my bad. Github emailed me and I assumed that meant you had requested a review. Let me see if I can muck with the settings...